util/settings/settings.go

```diff
@@ -42,6 +42,7 @@ type ArgoCDSettings struct {
 	// Indicates if status badge is enabled or not.
 	StatusBadgeEnabled bool `json:"statusBadgeEnable"`
 	// Admin superuser password storage
+	DisableAdmin       bool      `json:"disableAdmin,omitempty"`
```

Comment:
The `disableAdmin` field is not stored in settings. I think we should not use SettingManager to expose it to SessionManager and settings API. Instead value should be passed directly to SessionManager,  setting API server and  InitializeSettings method.